Remotion/Web/Development.WebTesting.UnitTests/Accessibility/AccessibilityAnalyzerTest.cs

```diff
@@ -0,0 +1,273 @@
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using log4net;
+using Remotion.Web.Development.WebTesting.Accessibility;
+using Remotion.Web.Development.WebTesting.Accessibility.Implementation;
+using Rhino.Mocks;
+
+namespace Remotion.Web.Development.WebTesting.UnitTests.Accessibility
+{
+  [TestFixture]
+  public class AccessibilityAnalyzerTest
+  {
+    private IAxeSourceProvider _sourceProviderStub;
+    private IAxeResultParser _resultParserStub;
+    private IAccessibilityResultMapper _resultMapperStub;
+    private IAccessibilityConfiguration _configStub;
+    private IJavaScriptExecutor _jsExecutorStub;
+    private IWebDriver _webDriverStub;
+    private ILog _loggerStub;
+
+    [SetUp]
+    public void SetUp ()
+    {
+      _jsExecutorStub = MockRepository.GenerateStub<IJavaScriptExecutor>();
+      _sourceProviderStub = MockRepository.GenerateStub<IAxeSourceProvider>();
+      _resultParserStub = MockRepository.GenerateStub<IAxeResultParser>();
+      _configStub = MockRepository.GenerateStub<IAccessibilityConfiguration>();
+      _webDriverStub = MockRepository.GenerateStub<IWebDriver>();
+      _resultMapperStub = MockRepository.GenerateStub<IAccessibilityResultMapper>();
+      _loggerStub = MockRepository.GenerateStub<ILog>();
+    }
+
+    [Test]
+    public void CreateAnalyzer_CopiesGivenConfig ()
+    {
+      _configStub
+          .Stub (_ => _.IFrameTimeout)
+          .Return (TimeSpan.FromSeconds (5));
+      _configStub
+          .Stub (_ => _.IncludeIFrames)
+          .Return (true);
+
+      var analyzer = new TestableAccessibilityAnalyzer (
+          _webDriverStub,
+          _jsExecutorStub,
+          _resultParserStub,
+          _configStub,
+          _sourceProviderStub,
+          _resultMapperStub,
+          _loggerStub);
+
+      Assert.That (analyzer.IncludeIFrames, Is.True);
+      Assert.That (analyzer.IFrameTimeout, Is.EqualTo (TimeSpan.FromSeconds (5)));
+    }
+
+    [Test]
+    public void CreateAnalyzer_ChangesDefaultConfiguration ()
+    {
+      _configStub
+          .Stub (_ => _.IFrameTimeout)
+          .Return (TimeSpan.FromSeconds (5));
+      _configStub
+          .Stub (_ => _.IncludeIFrames)
+          .Return (true);
+
+      var analyzer = new TestableAccessibilityAnalyzer (
```

Comment:
I think we should have a factory method. this is a lot of duplication maintenance.